src/codegen/operator/index_scan_translator.cpp

```diff
@@ -0,0 +1,343 @@
+    } else {
+      output_col_ids.resize(table.GetSchema()->GetColumnCount());
+      std::iota(output_col_ids.begin(), output_col_ids.end(), 0);
+    }
+    //    const auto &output_col_ids = index_scan_.GetColumnIds();
+
+    for (oid_t col_idx = 0; col_idx < output_col_ids.size(); col_idx++) {
+      final_attribute_accesses.emplace_back(tile_group_access,
+                                            final_ais[output_col_ids[col_idx]]);
+    }
+    for (oid_t col_idx = 0; col_idx < output_col_ids.size(); col_idx++) {
+      auto *attribute = final_ais[output_col_ids[col_idx]];
+      final_batch.AddAttribute(attribute, &final_attribute_accesses[col_idx]);
+    }
+
+    ConsumerContext context{this->GetCompilationContext(), this->GetPipeline()};
+    context.Consume(final_batch);
+
+    // Move to next tuple in the index scan result
+    result_iter = codegen->CreateAdd(result_iter, codegen.Const64(1));
+    loop.LoopEnd(codegen->CreateICmpULT(result_iter, result_size),
+                 {result_iter});
+  }
+
+  // free the memory allocated for the index scan iterator
+  codegen.Call(RuntimeFunctionsProxy::DeleteIterator, {iterator_ptr});
+}
+
+// Get the name of this scan
+std::string IndexScanTranslator::GetName() const {
+  std::string name = "Scan('" + GetIndex().GetName() + "'";
+  auto *predicate = GetIndexScanPlan().GetPredicate();
+  if (predicate != nullptr && predicate->IsSIMDable()) {
+    name.append(", ").append(std::to_string(Vector::kDefaultVectorSize));
+  }
+  name.append(")");
+  return name;
+}
+
+// Index accessor
+const index::Index &IndexScanTranslator::GetIndex() const {
+  return dynamic_cast<index::Index &>(*index_scan_.GetIndex().get());
+}
+
+void IndexScanTranslator::UpdateTupleWithParameterCache(
+    CodeGen &codegen, llvm::Value *iterator_ptr) const {
+  std::vector<const planner::AttributeInfo *> where_clause_attributes;
+  std::vector<ExpressionType> comparison_type;
+  const auto *predicate = index_scan_.GetPredicate();
+  if (predicate != nullptr) {
+    auto &context = GetCompilationContext();
+    const auto &parameter_cache = context.GetParameterCache();
+    predicate->GetUsedAttributesInPredicateOrder(where_clause_attributes);
+    predicate->GetComparisonTypeInPredicateOrder(comparison_type);
+    for (unsigned int i = 0; i < where_clause_attributes.size(); i++) {
+      const auto *ai = where_clause_attributes[i];
+      llvm::Value *attribute_id = codegen.Const32(ai->attribute_id);
+      llvm::Value *attribute_name = codegen.ConstStringPtr(ai->name);
+      bool is_lower_key = false;
+      if (comparison_type[i] == peloton::ExpressionType::COMPARE_GREATERTHAN ||
+          comparison_type[i] ==
+              peloton::ExpressionType::COMPARE_GREATERTHANOREQUALTO) {
+        is_lower_key = true;
+      }
+      llvm::Value *is_lower = codegen.ConstBool(is_lower_key);
+
+      llvm::Value *parameter_value = parameter_cache.GetValue(i).GetValue();
+      switch (ai->type.type_id) {
```

Comment:
Can we update the tuple value using `table_storage.cpp` so that we can remove these function calls and also the overhead setting the value through non-codegen'ed type system?  Obviously, as we talked before, using the tuple format is not the correct way to handle this and to be refactorized.  For this PR, this would be enough.